docs/comparison-matrix.md

```diff
@@ -31,8 +31,72 @@ Legend:
 | Language specification                                 | 🟢                | 🟢           | 🟢              | 🟢                 | 🟢       | 🟢           |
 | Backwards compatibility strategy                       | 🟢                | 🔴           | 🔴              | 🔴                 | 🔴       | 🔴           |
 
-:todo: add chapter elaborating on each feature
-## References 
+## Descriptions
+
+### Sum Types
+
+Types of the form `Time = Present | Past | Future`, which allow a type do be
+constructed by one of many variants. Think Rust's `enums`.
+
+### Product Types
+
+Types of the form `Person = MkPerson Age Name`, where `MkPerson` is of Kind
+`Type->Type->Type`. Product types combine multiple elements into one data type
+without tagging the elements.
+
+### Record Types
+
+Types of the form `Person = MkPerson { age :: Age, name :: Name }`. Record types
+are similar to `structs` in most programming languages.
+
+### Recursive Types
+
+Recursive types are defined by the presence of the LHS type in its RHS
+definition. A classic example is:
+
+```text
+List a = Nil | Cons a (List a)
+^^^^^^                 ^^^^^^
+```
+
+### Type Functions (Generics)
```

Comment:
Perhaps it's better what @gnumonik suggested. `Parameterized Types (Generics)`?

Reply:
```suggestion
### Parameterized Types (Generics)
```